Add more detail around the set up steps

tahb opened the pull request:
# Changes in this PR

This started out with the intention of documenting testing steps based on Pat's feedback. From working my way through the set up I felt there was enough value to add and get feedback whilst I continue looking at the testing steps.

# Release checklist

As part of our continuous deployment strategy we must ensure that this work is
ready to be released at any point. Before merging to `main` we must first
confirm:

## Pre merge checklist

- [ ] Have all changes to any dependencies been deployed to both `preprod` and
    `production` in a backwards compatible way? (This includes our API,
    infrastructure, third-party integrations and any secrets or environment variables)
- [ ] Has a required data migration been included in this change or been done in
    advance?

## Post merge checklist

Once we've merged we now need to release this through to production before
considering it done.

[Find the build-and-deploy job in CircleCI](https://app.circleci.com/pipelines/github/ministryofjustice/hmpps-temporary-accommodation-ui)
and work through the following steps:

- [ ] Has the product manager asked to provide approval for this feature?
  - [ ] Has the product manager approved?
- [ ] Manually approve release to preprod
- [ ] Verify change released to preprod
- [ ] Manually approve release to prod
- [ ] Verify change released to production

Should a release fail at any step, you as the author should now lead the work to
fix it as soon as possible.

tahb committed:
Document environments

tahb committed:
Move docker testing set up until just before it's needed

tahb committed:
Run shellcheck as part of code linting

tahb committed:
Update .env.example file for new joiners

By default set the API to run against the real server rather than Wiremock.

tahb committed:
Format markdown to 80 chars per line

tahb committed:
Update readme steps for getting running

- Include the preference of using ap tools plus a little of the reasons why
- Document wiremock and how you can configure it

tahb commented on package.json:
```diff
@@ -18,7 +18,7 @@
     "start-feature:dev": "npm run build && concurrently -k -p \"[{name}]\" -n \"Views,TypeScript,Node,Sass\" -c \"yellow.bold,cyan.bold,green.bold,blue.bold\" \"npm run watch-views\" \"npm run watch-ts\" \"npm run watch-node-feature\" \"npm run watch-sass\"",
     "record-build-info": "node ./bin/record-build-info",
-    "lint": "eslint . --cache --max-warnings 0",
+    "lint": "eslint . --cache --max-warnings 0 && node_modules/.bin/shellcheck ./script/**/*[^utils] ./script/utils/**",
```
I hope this is not breaking some unseen rule! I couldn't see any justification for it being a separate command in script/test but having it hear made sense? Happy to drop if not.

bryangaledxw replied:
I usually run `npm run lint -- --fix`, which I think this will break. Able to add something like `lint:fix`, to do the same as above but with the first bit changed to `eslint . --cache --max-warnings 0 --fix`?

tahb replied:
Sure 👍

tahb commented on .env.example:
```diff
@@ -1,5 +1,15 @@
+ENVIRONMENT=local
```
Just incase you want to use/test Sentry or anything else we use this for.

tahb commented on .env.example:
```diff
@@ -1,5 +1,15 @@
+COMMUNITY_ACCOMMODATION_API_SERVICE_NAME=temporary-accommodation
+WIREMOCK_ENDPOINT=http://localhost:9092
+
+# -- Wiremock on
```
I still find these comments useful, do you? Happy to drop if not.

bryangaledxw replied:
Hmm, I'm not even sure WIREMOCK_ENDPOINT does anything? I can't find any reference to it in code. And I generally switch between running via `script/server` (wiremock) and `ap-tools` (full stack) without needing to touch this. Are we sure these still do anything?

tahb replied:
You'll know better than I 👍 I'll drop this.

tahb commented on README.md:
```diff
@@ -27,6 +27,46 @@ script/bootstrap
+### 1. Using AP Tools
+
+In order to spin up a full stack of a working API and other [dependant
+services](./docker-compose.yml) we recommend using the [AP
```
I took the liberty of proposing this as our position without discussing it so let's take this opportunity to discuss and disagree/agree with it?

bryangaledxw commented on README.md:
```diff
@@ -81,3 +81,7 @@ The steps are also available in the [PULL_REQUEST_TEMPLATE](/.github/PULL_REQUES
+
+## Environments
+
+[Details of the different environments and their roles can be found in Confluence](https://dsdmoj.atlassian.net/wiki/spaces/AP/pages/4330226204/Environments).
```
My preference would be for just "can be found in Confluence" or even just "Confluence" to be the link

tahb replied:
I think I am happy to keep a sentence here to describe why you might want to dig into this.

bryangaledxw commented on .env.example:
```diff
@@ -1,5 +1,15 @@
-COMMUNITY_ACCOMMODATION_API_SERVICE_NAME=approved-premises
+SERVICE_SIGNIFIER=path
+APPROVED_PREMISES_SUBDOMAIN=approved-premises
+TEMPORARY_ACCOMMODATION_SUBDOMAIN=temporary-accommodation
```
I think these aren't used any more

bryangaledxw commented on .env.example:
```diff
@@ -1,5 +1,15 @@
+TEMPORARY_ACCOMMODATION_SUBDOMAIN=temporary-accommodation
 APPROVED_PREMISES_ROOT_PATH=approved-premises
 TEMPORARY_ACCOMMODATION_ROOT_PATH=temporary-accommodation
+COMMUNITY_ACCOMMODATION_API_SERVICE_NAME=temporary-accommodation
```
What would you think of getting rid of this from .env.example, and updating the default in config, which is `approved-premises`?

bryangaledxw commented on README.md:
````diff
@@ -27,6 +27,46 @@ script/bootstrap
+```
+
+```bash
+ap-tools server start --local-ui --local-api
````
Is this meant to be `ap-tools server stop`

bryangaledxw commented on README.md:
```diff
@@ -27,6 +27,46 @@ script/bootstrap
+tools](https://github.com/ministryofjustice/hmpps-approved-premises-tools#start-server).
+
+### 2. Manually
+
```
Worth a small note about why we might do this? - Quicker initial startup time, and able to test features that aren't yet present in the API?

bryangaledxw commented on README.md:
```diff
@@ -27,6 +27,46 @@ script/bootstrap
+
+For either option the frontend can further decide if it's requests should be
+sent to the locally running API or to a wiremock version. To do this we change
+the environment variable `APPROVED_PREMISES_API_URL` within .env to either of
```
Flagging again that I never seem to need to touch this

bryangaledxw reviewed (commented):
Definitely worthwhile cleanup! Think maybe this is adding some cruft into `.env.example` and the instructions though? Happy to chat if you disagree with my comments

tahb committed:
Hardcode service name

Given we don't share this frontend with CAS1 anymore we can hardcode
this value instead.

bryangaledxw reviewed (approved):
LGTM. Should the second-to-last commit message be "Given we don't share a frontend let's try and hard code it rather than passing it in."?

tahb commented:
Ha yes I will make that message make sense 🙈
